### DISCUSSION TITLE
Riverpod and flutter hooks in flutter 3.0 shows warning

### DISCUSSION THREAD

**Discussion by 2kjm** [rrousselGit/riverpod#1406] [Q&A] [answered]
While running the my app I am getting the following warning. What causes this?
`
../../../CORE/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_riverpod-1.0.3/lib/src/framework.dart:275:26: Warning: Operand of
null-aware operation '!' has type 'SchedulerBinding' which excludes null.
 - 'SchedulerBinding' is from 'package:flutter/src/scheduler/binding.dart'
 ('../../../CORE/flutter/packages/flutter/lib/src/scheduler/binding.dart').
    if (SchedulerBinding.instance!.schedulerPhase ==
                         ^
../../../CORE/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_hooks-0.18.3/lib/src/widgets_binding_observer.dart:42:29: Warning: Operand
of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart'
 ('../../../CORE/flutter/packages/flutter/lib/src/widgets/binding.dart').
    _state = WidgetsBinding.instance!.lifecycleState;
                            ^
../../../CORE/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_hooks-0.18.3/lib/src/widgets_binding_observer.dart:43:20: Warning: Operand
of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart'
 ('../../../CORE/flutter/packages/flutter/lib/src/widgets/binding.dart').
    WidgetsBinding.instance!.addObserver(this);
                   ^
../../../CORE/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_hooks-0.18.3/lib/src/widgets_binding_observer.dart:52:20: Warning: Operand
of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart'
 ('../../../CORE/flutter/packages/flutter/lib/src/widgets/binding.dart').
    WidgetsBinding.instance!.removeObserver(this);`

**Comment by rrousselGit** [comment, marked as answer]
Try running `flutter pub upgrade`

**Reply by 2kjm** [reply]
Wow. Answer in 12 seconds. Thanks, mate. It worked.

**Reply by yogithesymbian** [reply]
cool man , work like a charm.
